Applications/Spire/Source/KeyBindings/OrderTaskArgumentsMatch.cpp

```diff
@@ -0,0 +1,177 @@
+#include "Spire/KeyBindings/OrderTaskArgumentsMatch.hpp"
+#include "Nexus/Definitions/DefaultDestinationDatabase.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+
+using namespace boost;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  bool matches_by_word(const QString& text, const QString& query) {
+    auto words = text.splitRef(' ', Qt::SkipEmptyParts);
+    for(auto& word : words) {
+      if(text.midRef(
+          text.indexOf(word)).startsWith(query, Qt::CaseInsensitive)) {
+        return true;
+      }
+    }
+    return false;
+  }
+}
+
+bool Spire::matches(const QString& name, const QString& query) {
+  for(auto i = 0; i < name.length(); ++i) {
+    if(name.right(name.length() - i).startsWith(query, Qt::CaseInsensitive)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool Spire::matches(const CountryCode& country, const QString& query) {
+  auto& entry = GetDefaultCountryDatabase().FromCode(country);
```

Comment:
The default databases are for testing only, they can not be used in implementations.

Reply:
How to get the name, letter code of a country using country code?

Reply:
Pass in a ```CountryDatabase```.